tests/e2e/test_rosacli_operator_roles.go

```diff
@@ -1080,3 +1080,132 @@ var _ = Describe("Create cluster with oprator roles which are attaching managed
 				Expect(err).To(BeNil(), "It met error or timeout when waiting cluster to installing status")
 			})
 	})
+
+var _ = Describe("Upgrade operator roles in auto mode",
+	labels.Feature.Cluster, func() {
+		defer GinkgoRecover()
+		var (
+			rosaClient         *rosacli.Client
+			clusterService     rosacli.ClusterService
+			customProfile      *profilehandler.Profile
+			ocmResourceService rosacli.OCMResourceService
+			clusterName        string
+			clusterID          string
+		)
+
+		BeforeEach(func() {
+			By("Init client and service")
+			rosaClient = rosacli.NewClient()
+			ocmResourceService = rosaClient.OCMResource
+			clusterService = rosaClient.Cluster
+
+			By("Prepare custom profile")
+			customProfile = &profilehandler.Profile{
+				ClusterConfig: &profilehandler.ClusterConfig{
+					STS: true,
+				},
+				AccountRoleConfig: &profilehandler.AccountRoleConfig{
+					Path:               "",
+					PermissionBoundary: "",
+				},
+				Version:      "y-1",
+				ChannelGroup: "candidate",
+				Region:       "us-west-2",
+			}
+			customProfile.NamePrefix = constants.DefaultNamePrefix
+		})
+
+		AfterEach(func() {
+			if clusterID != "" {
+				By("Delete cluster")
+				rosaClient.Runner.UnsetArgs()
+				_, err := clusterService.DeleteCluster(clusterID, "-y")
+				Expect(err).To(BeNil())
+
+				rosaClient.Runner.UnsetArgs()
+				err = clusterService.WaitClusterDeleted(clusterID, 3, 30)
+				Expect(err).To(BeNil())
+			}
+
+			By("Clean resource")
+			errs := profilehandler.DestroyResourceByProfile(customProfile, rosaClient)
+			Expect(len(errs)).To(Equal(0))
+		})
+
+		It("to create and upgrade operator roles in auto mode - [id:45745]",
```

Comment:
OCP-45745 on Polarion is described as the scenario that update account roles and operator-roles then upgrade cluster.
In you automation bellow, it seems missing the step to upgrade cluster after the roles upgrade.
And this scenarios we need to the account-roles and operator-roles versions are Y-1 before the cluster upgrade, that's the real customers scenario.

Reply:
updated according to comments.